### PR TITLE
Refine cancelMPPQuery

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: ref #5095

Problem Summary:

### What is changed and how it works?
1. Do not hold `MPPTask` in `cancelMPPQuery` thread, so `MPPTask` can be destructed parallelly
2. Use ThreadManager instead of raw threads when cancel mpp task

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Mpp/MPPTaskManager.cpp [hunk 1]
```diff
@@ -89,31 +97,22 @@ void MPPTaskManager::cancelMPPQuery(UInt64 query_id, const String & reason)
     LOG_WARNING(log, fmt::format("Begin cancel query: {}", query_id));
     FmtBuffer fmt_buf;
     fmt_buf.fmtAppend("Remaining task in query {} are: ", query_id);
-    // TODO: cancel tasks in order rather than issuing so many threads to cancel tasks
```
Comment (SeaRise) [inline dbms/src/Flash/Mpp/MPPTaskManager.cpp:92]: It looks like this annotation cannot be removed, or not using `thread_manager`

File: dbms/src/Flash/Mpp/MPPTaskManager.cpp [hunk 2]
```diff
@@ -89,31 +97,22 @@ void MPPTaskManager::cancelMPPQuery(UInt64 query_id, const String & reason)
...
-    for (const auto & task : task_set->task_map)
+    auto thread_manager = newThreadManager();
+    for (auto it = task_set->task_map.begin(); it != task_set->task_map.end();)
     {
-        fmt_buf.fmtAppend("{} ", task.first.toString());
-        std::thread t(&MPPTask::cancel, task.second, std::ref(reason));
-        cancel_workers.push_back(std::move(t));
+        fmt_buf.fmtAppend("{} ", it->first.toString());
+        auto current_task = it->second;
+        it = task_set->task_map.erase(it);
```
Comment (SeaRise) [inline dbms/src/Flash/Mpp/MPPTaskManager.cpp:105]: Looks like `isTaskToBeCancelled` needs to be modified too
```
bool MPPTaskManager::isTaskToBeCancelled(const MPPTaskId & task_id)
{
    std::unique_lock lock(mu);
    auto it = mpp_query_map.find(task_id.start_ts);
    if (it != mpp_query_map.end() && it->second->to_be_cancelled)
    {
        return it->second->task_map.find(task_id) != it->second->task_map.end();
    }
    return false;
}
```

Comment (windtalker) [reply dbms/src/Flash/Mpp/MPPTaskManager.cpp:106]: Done

File: dbms/src/Flash/Mpp/MPPTaskManager.cpp (hunk 1 above)
Comment (windtalker) [reply dbms/src/Flash/Mpp/MPPTaskManager.cpp:92]: Add it back

**Review by yibin87 — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]

**Pull request merged**